app/src/context/evm/prompt-wars-market-contract/PromptWarsMarketContractContextController.tsx

```diff
@@ -0,0 +1,377 @@
+            fees,
+            management,
+            collateralToken,
+            // TODO:
+            outcomeIds: [],
+            isResolved: false,
+            isOpen: false,
+            isOver: false,
+            isRevealWindowExpired: false,
+            isResolutionWindowExpired: false,
+            isExpiredUnresolved: false,
+            status: PromptWarsMarketContractStatus.LOADING,
+          };
+
+          const status = getMarketStatus(values);
+
+          values.status = status;
+
+          setMarketContractValues(values);
+        } catch (error) {
+          console.log(error);
+        }
+      }, 1000);
+    } catch {
+      toast.trigger({
+        variant: "error",
+        withTimeout: true,
+        title: "Failed to fetch market data",
+        children: <Typography.Text>Try refreshing the page, or check your internet connection.</Typography.Text>,
+      });
+    }
+
+    setActions((prev) => ({
+      ...prev,
+      fetchMarketContractValues: {
+        isLoading: false,
+      },
+    }));
+  };
+
+  // const getOutcomeToken = async (args: GetOutcomeTokenArgs) => {
+  //   try {
+  //     if (!signer) throw new Error("not signed in");
+
+  //     if (!marketContract) {
+  //       marketContract = await await connect(DEPLOYED_CONTRACT_ADDRESS, signer);
+  //     }
+
+  //     const outcomeToken = await marketContract.get_outcome_token(args);
+
+  //     return outcomeToken;
+  //   } catch (error) {
+  //     console.log(error);
+  //   }
+
+  //   return undefined;
+  // };
+
+  const sell = async () => {
+    try {
+      assertWalletConnection();
+
+      await marketContract.sell();
+
+      toast.trigger({
+        variant: "confirmation",
+        withTimeout: false,
+        title: "Success",
+        children: <Typography.Text>Check your new wallet balance.</Typography.Text>,
+      });
+    } catch {
+      toast.trigger({
+        variant: "error",
+        withTimeout: true,
+        title: "Failed to call sell method",
+        children: (
+          <Typography.Text>Check your internet connection, your wallet connection and try again.</Typography.Text>
+        ),
+      });
+    }
+  };
+
+  const ftTransferCall = async (prompt: string) => {
+    if (!marketContractValues) {
+      return;
+    }
+
+    try {
+      assertWalletConnection();
+
+      setActions((prev) => ({
+        ...prev,
+        ftTransferCall: {
+          ...prev.ftTransferCall,
+          isLoading: true,
+        },
+      }));
+
+      const amount = marketContractValues.fees.price.toString();
+      const msg = JSON.stringify({ CreateOutcomeTokenArgs: { prompt: JSON.stringify(prompt) } });
+
+      // TODO: What is fungible contract
+      // await FungibleTokenContract.ftTransferCall(
+      //   walletStateContext.context.wallet!,
+      //   marketContractValues.collateralToken.id!,
+      //   {
+      //     receiver_id: marketId,
+      //     amount,
+      //     msg,
+      //   },
+      // );
+
```

Comment:
What is fungible contract for? and there is no ft_transfer_call method in Market.sol contract

Reply:
Solidity ERC20 architecture is different in this way. 
The new flow for a user to play is to create 2 transactions: 

1. is to submit an ERC20 allowance for the price of the game:

```
const marketSpendingAllowance = await collateralToken.allowance(
      player.address,
      market.address
    );
```

2. Then the player should be able to call the `register` on `Market.sol`